lms/templates/problem.html

```diff
@@ -14,13 +14,13 @@ <h2 class="problem-header">
     <input type="hidden" name="problem_id" value="${ problem['name'] }" />
 
     % if check_button:
-    <input class="check ${ check_button }" type="button" data-checking="${ check_button_checking }" value="${ check_button }" />
+    <button class="check ${ check_button }" data-checking="${ check_button_checking }" data-value="${ check_button }"><span class="check-label">${ check_button }</span><span class="sr"> ${_("your answer")}</span></button>
     % endif
     % if reset_button:
-    <input class="reset" type="button" value="${_('Reset')}" />
+    <button class="reset" data-value="${_('Reset')}">${_('Reset')}<span class="sr"> ${_("your answer")}</span></button>
     % endif
     % if save_button:
-    <input class="save" type="button" value="${_('Save')}" />
+    <button class="save" data-value="${_('Save')}">${_('Save')}<span class="sr"> ${_("your answer")}</span></button>
     % endif
     % if answer_available:
     <button class="show"><span class="show-label">${_('Show Answer')}</span> <span class="sr">${_("Reveal Answer")}</span></button>
```

Comment:
@ahsan-ul-haq, I'm jsut wondering what the data-* attributes are doing on the Check, Reset, and Save buttons and if they should be added to the Show Answer button as well.  I also think the SR text "Reveal Answer" is unnecessary here since the button value is so similar "Show Answer".  As it stands, the screen reader user would hear "Show Answer, Reveal Answer".  I say remove the SR text here.